paddle/fluid/pybind/imperative.cc

```diff
@@ -467,20 +479,25 @@ static void ParseIndexingSlice(framework::LoDTensor *tensor, PyObject *_index,
       slice_ends->push_back(start + 1);
       slice_strides->push_back(1);
       decrease_axis->push_back(dim);
-    } else {
+      dim++;
+    } else if (PySlice_Check(slice_item)) {
       // slice item
       Py_ssize_t start, end, step;
       PySliceObject *p = reinterpret_cast<PySliceObject *>(slice_item);
       _PySlice_GetIndices(p, dim_len, &start, &end, &step);
 
       // :: or : or 0:dim_len:1
       if (start == 0 && end == dim_len && step == 1) {
+        dim++;
         continue;
       }
       slice_axes->push_back(dim);
       slice_starts->push_back(start);
       slice_ends->push_back(end);
       slice_strides->push_back(step);
+      dim++;
+    } else if (slice_item == Py_Ellipsis) {
+      dim += rank - specified_dims;
```

Comment:
may be adding else branch is better, we can move the error hint into else branch, such as
```
} else {
PADDLE_THROW(
        platform::errors::InvalidArgument(
                  "Currently, VarBase.__getitem__() only allows "
                  "indexing by Integers, Slices, Ellipsis, and tuples of "
                  "these types, but received %s in %dth slice item",
                  std::string(Py_TYPE(slice_item)->tp_name), i + 1));
}
```

Reply:
Done!